### ISSUE TITLE
[question] How to debug private dependency issues

### ISSUE THREAD

**Issue by sunj1** [conan-io/conan#12657]
Hello,

I am using conan version 1.52

I have a library package, say Lib1 having option SocketType which can take values S1,S2,S3, etc.
I have components which use this Lib1 package as a dependency.
So, I have Comp1 which requires self.options['Lib1'].SocketType=S1
Also, I have another component Comp2, which requires a different value of SocketType.

Below are excerpts from my configuration in short:

conanfile of Comp1 looks like:
```
self.requires("Lib1", private=True)

self.options["Lib1"].SocketType = "S1"
```

conanfile of Comp2 looks like:
```
self.requires("Lib1", private=True)

self.options["Lib1"].SocketType = "S2"
```

conanfile of my main consumer package looks like:
```
self.requires("Comp1")
self.requires("Comp2")
self.requires("Lib1")

self.options["Lib1"].SocketType = "S2"
```

Now, when I try to run conan info command, I get error saying:  
`ERROR: Comp1 tried to change option SocketType to S1 but it was already assigned to S2 by MainConsumerPkg`

How do I proceed to debug the actual cause of the ERROR. I tried setting CONAN_LOGGING_LEVEL=1 but still I do not get information which will help me find the error in dependencies.

Could you please let me know some way by which I can see more verbose steps to figure out the issue?

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by memsharded** [comment]
Hi @sunj1 

Not sure what kind of messages you would be expecting, but it seems the one Conan is giving is quite complete: there are two different packages "Comp1" and "MainConsumerPkg" that are trying to assign different values to the same option in one dependency, and that is a conflict.

It is necessary to resolve the conflict, typically by specifying in the most downstream consumer the value of the option that you want.
Also, in the general case, it is better to try to avoid changing many options defaults in recipes, and changing them in profiles, because those changes are more global and more consistent.

The last ``self.options["Lib1"].SocketType = "S2"`` should in theory resolve the conflict, but it seems for some reason it is not. It would be better to provide complete recipes to reproduce the problem, otherwise it is very difficult to know what could be happening.

**Comment by sunj1** [comment]
Actually, since the Lib1 package is private in the Components Comp1 and Comp2,
the library Lib1 is not visible outside the Component packages.

Hence there should not be a conflict -- if I understood the private dependency properly in conan workflow.

**Comment by memsharded** [comment]
> Actually, since the Lib1 package is private in the Components Comp1 and Comp2,

It is totally possible to have that setup, if dependencies are private. Here is a test that proves it:
```python
    def test_conflict_transitive_option(self):
        t = TestClient()
        t.save({'top.py': GenConanfile("top", "v1").with_option("myopt", [1, 2]),
                'middle.py': GenConanfile("middle", "v1").with_require("top/v1", private=True)
                                                         .with_default_option("top*:myopt", 1),
                'middle2.py': GenConanfile("middle2", "v1").with_require("top/v1", private=True)
                                                           .with_default_option("top*:myopt", 2),
                'app.py': GenConanfile() .with_requires("middle/v1", "middle2/v1")
                })
        t.run("export top.py")
        t.run("export middle.py")
        t.run("export middle2.py")
        t.run("install app.py --build=missing")
        assert "top/v1:533491fbfea90f5db4918f2da7a9b6120b6df144 - Build" in t.out
        assert "top/v1:90e302ce56ee2c53d60f6b5c7c33f95092637eba - Build" in t.out
```

**Comment by memsharded** [comment]
Another quick tip: use packages always in lowercase, not uppercase, as it will be mandatory in Conan 2.0 to use always lowercase.

**Comment by sunj1** [comment]
> > Actually, since the Lib1 package is private in the Components Comp1 and Comp2,
> 
> It is totally possible to have that setup, if dependencies are private. Here is a test that proves it:
> 

Yes, I am able to achieve this setup in one set of packages, but not in another set of packages.  
Hence looking for a way to get some details/steps to figure out the cause of issue.

**Comment by memsharded** [comment]
Closing this ticket as outdated/legacy. Conan 2 removed ``private`` dependencies, and it has a new system of ``requirement traits``.
Please create new tickets for any further question or issue, thanks for the feedback.
